yt_dlp/extractor/s4c.py

```diff
@@ -48,15 +59,57 @@ def _real_extract(self, url):
                 'thirdParty': 'false',
                 'filename': filename,
             }, note='Downloading streaming urls JSON')['hls']
-        formats, subtitles = self._extract_m3u8_formats_and_subtitles(m3u8_url, video_id, 'mp4', m3u8_id='hls')
-
+        formats = self._extract_m3u8_formats(m3u8_url, video_id, 'mp4', m3u8_id='hls')
         return {
             'id': video_id,
             'formats': formats,
             'subtitles': subtitles,
+            'thumbnail': thumbnail,
             **traverse_obj(details, ('full_prog_details', 0, {
                 'title': (('programme_title', 'series_title'), {str}),
                 'description': ('full_billing', {str.strip}),
                 'duration': ('duration', {lambda x: int(x) * 60}),
             }), get_all=False),
         }
+
+
+class S4CSeriesIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:www\.)?s4c\.cymru/clic/series/(?P<id>\d+)'
+    _TESTS = [{
+        'url': 'https://www.s4c.cymru/clic/series/864982911',
+        'playlist_mincount': 6,
+        'info_dict': {
+            'id': '864982911',
+            'title': 'Iaith ar Daith',
+            'description': 'md5:e878ebf660dce89bd2ef521d7ce06397'
+        },
+    }, {
+        'url': 'https://www.s4c.cymru/clic/series/866852587',
+        'playlist_mincount': 8,
+        'info_dict': {
+            'id': '866852587',
+            'title': 'FFIT Cymru',
+            'description': 'md5:abcb3c129cb68dbb6cd304fd33b07e96'
+        },
+    }]
+
+    def _real_extract(self, url):
+        series_id = self._match_id(url)
+        seriesDetails = self._download_json(
+            'https://www.s4c.cymru/df/series_details', series_id, query={
+                'lang': 'e',
+                'series_id': series_id,
+                'show_prog_in_series': 'Y'
+            }, note='Downloading player config JSON')
+        episodes = (self.url_result(
+                    'https://www.s4c.cymru/clic/programme/%s' % episode['id'],
+                    video_id=episode['id'])
+                    for episode in traverse_obj(seriesDetails, ('other_progs_in_series')))
+
+        return self.playlist_result(
+            entries=episodes,
+            playlist_id=series_id,
+            **traverse_obj(seriesDetails, ('full_prog_details', 0, {
+                'title': (('series_title'), {str}),
+                'description': ('full_billing', {str.strip}),
+            }), get_all=False))
```

Comment:
Use `...` with traversal to branch and ensure that it will return a list
We don't need a whole `episode` dict, just the ID
F-strings are preferred
```suggestion

        return self.playlist_result(
            [self.url_result(f'https://www.s4c.cymru/clic/programme/{episode_id}', S4CIE, episode_id)
             for episode_id in traverse_obj(series_details, ('other_progs_in_series', ..., 'id'))],
            series_id, **traverse_obj(series_details, ('full_prog_details', 0, {
                'title': (('series_title'), {str}),
                'description': ('full_billing', {str.strip}),
            }), get_all=False))
```

Reply:
`                'title': (('series_title'), {str}),`

`('series_title')` ?

The `description` is that of one programme, apparently the latest shown, in the `series_details`, not that of the series itself, at least in the 2nd test. When a new episode is added the value may change. In this case, but not necessarily in general, the value from the first episode (`series_details['other_progs_in_series'][-1]['full_billing']`) is more appropriate. Probably just bin it.

Reply:
Thanks, I've edited my suggestion above